metricflow/time/time_granularity_solver.py

```diff
@@ -151,87 +72,65 @@ def validate_time_granularity(
 
         e.g. throw an error if "ds__week" is specified for a metric with a time granularity of MONTH.
         """
+        valid_group_by_elements = self._semantic_manifest_lookup.metric_lookup.linkable_set_for_metrics(
+            metric_references=metric_references,
+        )
+
         for time_dimension_spec in time_dimension_specs:
-            # Validate local time dimensions.
-            if time_dimension_spec.entity_links == ():
-                _, min_granularity_for_querying = self.local_dimension_granularity_range(
-                    metric_references=metric_references,
-                    local_time_dimension_reference=time_dimension_spec.reference,
+            match_found = False
+            for path_key in valid_group_by_elements.path_key_to_linkable_dimensions:
+                if (
+                    path_key.element_name == time_dimension_spec.element_name
+                    and (path_key.entity_links == time_dimension_spec.entity_links)
+                    and path_key.time_granularity == time_dimension_spec.time_granularity
+                ):
+                    match_found = True
+                    break
+            if not match_found:
+                raise RequestTimeGranularityException(
+                    f"{time_dimension_spec} is not valid for querying {metric_references}"
                 )
-                if time_dimension_spec.time_granularity < min_granularity_for_querying:
-                    raise RequestTimeGranularityException(
-                        f"The minimum time granularity for querying metrics {metric_references} is "
-                        f"{min_granularity_for_querying}. Got {time_dimension_spec}"
-                    )
-
-                # If there is a cumulative metric, granularity changes aren't supported. We need to check the granularity
-                # specified in the configs for the cumulative metric alone, since `min_granularity_for_querying` may not be supported.
-                for metric_reference in metric_references:
```

Comment:
Ah, cool, this was moved in to the valid dimensions accessor so we don't need it here.